gobblin-iceberg/src/test/java/org/apache/gobblin/iceberg/writer/GobblinMCEWriterTest.java

```diff
@@ -0,0 +1,250 @@
+
+  @Mock
+  private HiveTable mockTable;
+
+  @AfterMethod
+  public void clean() throws Exception {
+    gobblinMCEWriter.close();
+  }
+
+  @BeforeMethod
+  public void setUp() throws Exception {
+    initMocks();
+    gmceBuilder = GobblinMetadataChangeEvent.newBuilder()
+        .setDatasetIdentifier(DatasetIdentifier.newBuilder()
+            .setDataPlatformUrn("urn:namespace:dataPlatform:hdfs")
+            .setNativeName("testDB/testTable")
+            .build())
+        .setFlowId("testFlow")
+        .setSchemaSource(SchemaSource.EVENT)
+        .setOperationType(OperationType.add_files)
+        .setCluster(ClustersNames.getInstance().getClusterName());
+    watermark = new KafkaStreamingExtractor.KafkaWatermark(
+        new KafkaPartition.Builder().withTopicName("GobblinMetadataChangeEvent_test").withId(1).build(),
+        new LongWatermark(10L));
+
+    State state = new State();
+    String metadataWriters = String.join(",",
+        Arrays.asList(mockWriter.getClass().getName(), exceptionWriter.getClass().getName()));
+    state.setProp("gmce.metadata.writer.classes", metadataWriters);
+
+    Mockito.doNothing().when(mockWriter)
+        .writeEnvelope(
+            Mockito.any(RecordEnvelope.class), Mockito.anyMap(), Mockito.anyMap(), Mockito.any(HiveSpec.class));
+    Mockito.doThrow(new IOException("Test Exception")).when(exceptionWriter)
+        .writeEnvelope(
+            Mockito.any(RecordEnvelope.class), Mockito.anyMap(), Mockito.anyMap(), Mockito.any(HiveSpec.class));
+
+    PowerMockito.mockStatic(GobblinConstructorUtils.class);
+    when(GobblinConstructorUtils.invokeConstructor(
+            eq(MetadataWriter.class), eq(mockWriter.getClass().getName()), any(State.class)))
+        .thenReturn(mockWriter);
+    when(GobblinConstructorUtils.invokeConstructor(
+        eq(MetadataWriter.class), eq(exceptionWriter.getClass().getName()), any(State.class)))
+        .thenReturn(exceptionWriter);
+
+    PowerMockito.mockStatic(FileSystem.class);
+    when(FileSystem.get(any()))
+        .thenReturn(fs);
+
+    when(mockTable.getDbName()).thenReturn(dbName);
+    when(mockTable.getTableName()).thenReturn(tableName);
+    when(mockHiveSpec.getTable()).thenReturn(mockTable);
+
+    gobblinMCEWriter = new GobblinMCEWriter(new GobblinMCEWriterBuilder(), state);
+  }
+
+  @Test
+  public void testWriteWhenWriterSpecified() throws IOException {
+    gmceBuilder.setAllowedMetadataWriters(Arrays.asList(mockWriter.getClass().getName()));
+    writeToMetadataWriters(gmceBuilder.build());
+
+    Mockito.verify(mockWriter, Mockito.times(1)).writeEnvelope(
+        Mockito.any(RecordEnvelope.class), Mockito.anyMap(), Mockito.anyMap(), Mockito.any(HiveSpec.class));
+    Mockito.verify(exceptionWriter, never()).writeEnvelope(
+        Mockito.any(RecordEnvelope.class), Mockito.anyMap(), Mockito.anyMap(), Mockito.any(HiveSpec.class));
+  }
+
+  @Test
+  public void testFaultTolerance() throws IOException {
```

Comment:
This test kind of already exists in the Iceberg writer test but I felt the testing could be done more explicitly here without the fluff related to iceberg / hive